src/lib.rs

```diff
@@ -184,16 +167,22 @@ impl TimerFd {
     /// This can also fail in various cases of resource exhaustion. Please check
     /// `timerfd_create(2)` for details.
     pub fn new_custom(clock: ClockId, nonblocking: bool, cloexec: bool) -> IoResult<TimerFd> {
-
-        let mut flags = 0;
+        let mut flags = TimerfdFlags::empty();
         if nonblocking {
-            flags |= TFD_NONBLOCK;
+            flags |= TimerfdFlags::NONBLOCK;
         }
         if cloexec {
-            flags |= TFD_CLOEXEC;
+            flags |= TimerfdFlags::CLOEXEC;
         }
 
-        let fd = neg_is_err(unsafe { timerfd_create(clock as libc::c_int, flags) })?;
+        let clock = match clock {
+            ClockId::Realtime => TimerfdClockId::Realtime,
+            ClockId::RealtimeAlarm => TimerfdClockId::RealtimeAlarm,
+            ClockId::Monotonic => TimerfdClockId::Monotonic,
+            ClockId::Boottime => TimerfdClockId::Boottime,
+            ClockId::BoottimeAlarm => TimerfdClockId::BoottimeAlarm,
+        };
+        let fd = rustix::time::timerfd_create(clock as TimerfdClockId, flags)?;
```

Comment:
Isn't casting `as TimerfdClockId` redundant here?

Reply:
Yes, good catch. I've fixed this now.